src/connector/src/source/kafka/enumerator/client.rs

```diff
@@ -310,10 +314,10 @@ impl KafkaSplitEnumerator {
         let mut result = HashMap::with_capacity(partitions.len());
 
         for elem in offsets.elements_for_topic(self.topic.as_str()) {
+            println!("[rc] offset {:?}", elem.offset());
             match elem.offset() {
                 Offset::Offset(offset) => {
-                    // XXX(rc): currently in RW source, `offset` means the last consumed offset, so we need to subtract 1
-                    result.insert(elem.partition(), Some(offset - 1));
+                    result.insert(elem.partition(), Some(offset));
```

Comment:
Here why we need to `-1` before?
The doc of [rd_kafka_offsets_for_times](https://docs.confluent.io/platform/7.5/clients/librdkafka/html/rdkafka_8h.html#ae308aaf534807cfd4c71e423fc214929) said:
```
The returned offset for each partition is the earliest offset whose timestamp is greater than or equal to the given timestamp in the corresponding partition.
```
So I think we just need to start the consumption from the returned offset, not need to `-1`.

Reply:
Haha, your question is exactly the reason why I created this PR. See? No one can understand these -1/+1 without explanation or deep dive into the code. This is a maintenance hell! See https://github.com/risingwavelabs/risingwave/pull/16257#issuecomment-2060271006 for more explanation.

Reply:
I have read your [explanation](https://github.com/risingwavelabs/risingwave/pull/16257#issuecomment-2060271006) below. IIUC, we `-1` here to create a `Split`, then when the `Split` passed to source executor, it will `+1` to create a `Reader` start consumption, right?